jaxrs-clients/src/main/java/com/palantir/remoting2/jaxrs/feignimpl/FeignSerializableErrorErrorDecoder.java

```diff
@@ -42,7 +54,68 @@ public Exception decode(String methodKey, Response response) {
         } catch (IOException e) {
             return new RuntimeException("Cannot get input stream from response: " + e.getMessage(), e);
         }
-        return SerializableErrorToExceptionConverter.getException(contentTypes, response.status(), response.reason(),
+        Exception exception = SerializableErrorToExceptionConverter.getException(
+                contentTypes,
+                response.status(),
+                response.reason(),
                 body);
+
+        Collection<String> retryAfter =
+                HeaderAccessUtils.caseInsensitiveGet(response.headers(), HttpHeaders.RETRY_AFTER);
+        if (retryAfter != null) {
+            Date retryAfterDate = RETRY_AFTER_DECODER.apply(Iterables.getFirst(retryAfter, null));
+            if (retryAfterDate != null) {
+                return new RetryableException(exception.getMessage(), exception, retryAfterDate);
```

Comment:
It's used in the default Retryer `continueOrPropagate` and it sleeps until this time.